esp-hal/src/aes/mod.rs

```diff
@@ -154,8 +205,9 @@ impl<'d> Aes<'d> {
     }
 
     /// Encrypts/Decrypts the given buffer based on `mode` parameter
-    pub fn process(&mut self, block: &mut [u8; 16], mode: Mode, key: &[u8; 16]) {
-        self.write_key(key);
+    pub fn process(&mut self, block: &mut [u8; 16], mode: Mode, key: Key) {
```

Comment:
```suggestion
    pub fn process<K>(&mut self, block: &mut [u8; 16], mode: Mode, key: K)
    where
        K: Into<Key>,
    {
```

Maybe something like this would be preferable to having to add `.into()` everywhere whenever we invoke this? @MabezDev any reason not to do this that you can think of?

Reply:
I'm happy with what we have now, but maybe the  `Into` bound is better? In fact, if we add that, this won't be a breaking change in AES, right?

Reply:
Agree, just passing `keytext` is better for users than `keytext.into()`. Updated both code and comments

Reply:
> I'm happy with what we have now, but maybe the `Into` bound is better? In fact, if we add that, this won't be a breaking change in AES, right?

I guess it's not breaking change. In fact, it's even closer to initial (before this PR) function call, but just passing the value itself, not a reference (`&keytext` -> `keytext`)